packages/core/saved-objects/core-saved-objects-migration-server-internal/src/next.ts

```diff
@@ -242,6 +249,12 @@ export const nextActionMap = (
       }),
     MARK_VERSION_INDEX_READY: (state: MarkVersionIndexReady) =>
       Actions.updateAliases({ client, aliasActions: state.versionIndexReadyActions.value }),
+    MARK_VERSION_INDEX_READY_SYNC: (state: MarkVersionIndexReady) =>
```

Comment:
Here's where the synchronization magic happens:
* All migrators involved in a relocation will wait for the rest to reach this point.
* They will then provide the aliases that they intend to update (in the `payload` property).
* The `updateRelocationAliases` synchronisation object will `then` call the `client.indices.updateAliases` one single time.
* Each of the migrators will receive the same response for the global update.